src/systems.rs

```diff
@@ -321,25 +321,43 @@ pub(crate) fn particle_transform(
     mut particle_query: Query<(
         &Particle,
         &Lifetime,
-        &Direction,
+        &mut Velocity,
         &mut DistanceTraveled,
-        &mut Speed,
         &mut Transform,
     )>,
     time: Res<Time>,
 ) {
     particle_query.par_iter_mut().for_each_mut(
-        |(particle, lifetime, direction, mut distance, mut speed, mut transform)| {
+        |(particle, lifetime, mut velocity, mut distance, mut transform)| {
             let lifetime_pct = lifetime.0 / particle.max_lifetime;
-            if particle.use_scaled_time {
-                speed.0 +=
-                    particle.acceleration.at_lifetime_pct(lifetime_pct) * time.delta_seconds();
-                transform.translation += direction.0 * speed.0 * time.delta_seconds();
-            } else {
-                speed.0 +=
-                    particle.acceleration.at_lifetime_pct(lifetime_pct) * time.raw_delta_seconds();
-                transform.translation += direction.0 * speed.0 * time.raw_delta_seconds();
+
+            let delta_time = match particle.use_scaled_time {
+                true => time.delta_seconds(),
+                false => time.raw_delta_seconds(),
+            };
+
+            // The velocity_direction variable is needed due to its relation with the acceleration being a f32 and not a Vec3
+            let velocity_direction = velocity.0.normalize();
```

Comment:
Vector normalization is quite expensive, so having to do this for every particle for every frame would potentially perform pretty poorly at higher particle counts. Maybe we should just go all the way also replace `acceleration` with a vector as well so we can avoid the normalize call here.